### DISCUSSION TITLE
Exposed ports

### DISCUSSION THREAD

**Discussion by andreclemente** [music-assistant/support#2029] [Q&A] [answered]
Hi. I'm trying to move my Music Assistant to my TrueNAS, since I've tested it on my PC and it seems to work. 

Problem: 
I can't use the host network, since TrueNAS Scale limits exposed ports to > 9000
By doing the usual forward to some other port makes it possible to run Music Assistant.

However, **no player is shown** when using this forward, which makes me think that is some other port being used OR the exposed ports must be the 8095 and 8096 in order to make it work. The latest option doesn't make much sense to me, since I've tested changing the ports on my PC installation and it works (using host network).

Does anyone know how to make it work without using host network? 

Thank you in advance!

**Comment by OzGav** [comment]
Have you read the troubleshooting page in the docs?

**Reply by andreclemente** [reply]
Yes, I'm asking if "Ensure containers are in HOST networking mode" is mandatory or is there a way around it. 

As I said, if I try to do that, I get an error saying that port is already in use, even if I forward 8095 and 8096 or use other ports directly... So maybe it's using some ports that are not explicitly written in the docs that I also need to forward, idk.

**Reply by OzGav** [reply]
Yes it’s mandatory. If you are in host networking mode you shouldn’t have to forward any ports.

**Reply by OzGav** [reply]
I see you have some issue with your NAS config. Unfortunately we can’t help with that

**Reply by andreclemente** [reply]
I'm just asking if 8095 and 8096 are the only ports used by the music assistant server...

**Reply by OzGav** [reply, marked as answer]
No for a start slimproto needs 3483, 9000 and 9090. Snapcast https://music-assistant.io/player-support/snapcast/ 

There are lots. You can search discord to see various ports mentioned in posts

**Reply by marcelveldt** [reply]
Besides that you have mdns and upnp needing the host network and some of the providers select random ports. You need host networking or a bridge
